File: android/app/src/main/java/com/hindtechgroup/hindpharma/ViewModel/ManagerDashboardViewModel.kt
package com.hindtechgroup.hindpharma.ViewModel

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.hindtechgroup.hindpharma.Helper.ApiClient
import kotlinx.coroutines.launch

class ManagerDashboardViewModel : ViewModel() {
    var employeeCount by mutableStateOf(0); private set
    var medicalCount by mutableStateOf(0); private set
    var productCount by mutableStateOf(0); private set
    var isLoading by mutableStateOf(false); private set
    var errorMessage by mutableStateOf<String?>(null); private set
    fun load(token: String?) { if (isLoading) return; viewModelScope.launch { isLoading=true; errorMessage=null; try { val api=ApiClient(token); val users=api.getUsers(); employeeCount=users.count { it.role=="employee" && it.isActive }; medicalCount=api.getMedicals().size; productCount=api.getProducts().size } catch(e:Exception){ errorMessage=e.message?:"Dashboard could not be loaded." } finally { isLoading=false } } }
}

File: ios/HindPharmaApp/ViewModel/ManagerDashboardViewModel.swift
import Foundation
import Combine
@MainActor final class ManagerDashboardViewModel: ObservableObject {
 @Published var employeeCount=0; @Published var medicalCount=0; @Published var productCount=0; @Published var isLoading=false; @Published var errorMessage:String?
 func load(token:String?) async { guard !isLoading else{return}; isLoading=true; defer{isLoading=false}; do { let api=APIClient(token:token); async let u=api.users(); async let m=api.medicals(); async let p=api.products(); employeeCount=(try await u).filter{$0.role=="employee" && $0.is_active==1}.count; medicalCount=(try await m).count; productCount=(try await p).count } catch { errorMessage=error.localizedDescription } }
}